examples/simple_project/versions.tf

```diff
@@ -19,7 +19,7 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = "~> 3.30"
+      version = ">= 3.50, < 5.0"
```

Comment:
Lets bump these to `~> 4.5` here and throughout since that will be the min supported provider version

Reply:
Just to be explicit: if we expose this feature at the top level it imposes a 4.5+ requirement. you're willing to do that?

Reply:
>  if we expose this feature at the top level it imposes a 4.5+ requirement

The budget submodule [already](https://github.com/tpdownes/terraform-google-project-factory/blob/9ef4da545aaa6240664553eb900b84f0d79ac8de/modules/budget/versions.tf#L23) has 4.5+ requirement which would then add that as a version constraint to the [top level](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/40ba6da63e7b6d548ceee8235361ed96bf8517e5/main.tf#L88-L99). For instance, in the [minimal fixture](https://github.com/terraform-google-modules/terraform-google-project-factory/tree/master/test/fixtures/minimal), this would be evaluated to 
```
Finding hashicorp/google versions matching ">= 3.50.*, < 5.0.*, >= 3.50.*, < 5.0.*, >= 4.5.*, < 5.0.*, >= 3.43.*, < 5.0.*, >= 3.50.*, < 5.0.*, >= 3.43.*, < 5.0.*, >= 3.43.*, < 5.0.*"
...
Installing hashicorp/google v4.6.0...
 ```

Reply:
I see your point. I had been deliberately trying to impose that version on the top-level module but didn't realize (or look hard enough) that it imposes that requirement. Since it does, yeah, I agree entirely that we should just expose the variable through the top-level.

Reply:
On `HEAD`, I believe that I've updated every submodule, test, and example to use `~> 4.5` if they touch the budget features.